dev/ProgressBar/ProgressBar.cpp

```diff
@@ -93,32 +94,47 @@ void ProgressBar::UpdateStates()
     }
     else if (!IsIndeterminate())
     {
-        SetProgressBarIndicatorWidth();
         winrt::VisualStateManager::GoToState(*this, s_DeterminateStateName, true);
     }
 }
 
 void ProgressBar::SetProgressBarIndicatorWidth()
 {
+    const auto templateSettings = winrt::get_self<::ProgressBarTemplateSettings>(TemplateSettings());
+
     if (auto&& progressBar = m_layoutRoot.get())
     {
         if (auto&& progressBarIndicator = m_progressBarIndicator.get())
         {
             const double progressBarWidth = progressBar.ActualWidth();
+            const double prevIndicatorWidth = progressBarIndicator.ActualWidth();
             const double maximum = Maximum();
             const double minimum = Minimum();
             const auto padding = Padding();
 
-            if (std::abs(maximum - minimum) > DBL_EPSILON)
+            m_isUpdating = true; // Adds "Updating" state in between to trigger RepositionThemeAnimation Visual Transition in ProgressBar.xaml when reverting back to previous state
+            UpdateStates();
```

Comment:
>UpdateStates [](start = 12, length = 12)

Does this mean that if we were in the Error state, then we change the value that we will pop into the updating state for a second and then revert back to the error state? Is that intended?

Reply:
`Updating` state does not occur between `Paused` and `Error` state, since those `SetProgressBarIndicatorWidth()` is not called for those two states.